tsdownsample/downsamplers.py

```diff
@@ -1,39 +1,93 @@
+class MinMaxDownsampler(AbstractRustDownsampler):
+    def __init__(self) -> None:
+        super().__init__("MinMax", _tsdownsample_rs.minmax, rust_dtypes)
 
-MeanDownsampler = FuncDownsamplingInterface("Mean", np.mean)
-MedianDownsampler = FuncDownsamplingInterface("Median", np.median)
+    def _downsample(self, x: Union[np.ndarray, None], *args, **kwargs):
+        if x is not None:
+            warnings.warn(
+                f"x is passed to downsample method of {self.name}, but is not taken "
+                f"into account by the current implementation of  {self.name} algorithm."
+            )
+        return super()._downsample(None, *args, **kwargs)
 
-# ------------------ EveryNth Downsampler ------------------
-import math
 
-import pandas as pd
+class M4Downsampler(AbstractRustDownsampler):
+    def __init__(self):
+        super().__init__("M4", _tsdownsample_rs.m4, rust_dtypes)
+
+    def _downsample(self, x: Union[np.ndarray, None], *args, **kwargs):
+        if x is not None:
+            warnings.warn(
+                f"x is passed to downsample method of {self.name}, but is not taken "
+                f"into account by the current implementation of  {self.name} algorithm."
+            )
+        return super()._downsample(None, *args, **kwargs)
+
 
-from .downsampling_interface import DownsampleInterface
+class LTTBDownsampler(AbstractRustDownsampler):
+    def __init__(self):
+        super().__init__("LTTB", _tsdownsample_rs.lttb, rust_dtypes)
 
 
-class _EveryNthDownsampler(DownsampleInterface):
+class MinMaxLTTBDownsampler(AbstractRustDownsampler):
+    def __init__(self):
+        super().__init__("MinMaxLTTB", _tsdownsample_rs.minmaxlttb, rust_dtypes)
+
+    def downsample(
+        self, *args, n_out: int, minmax_ratio: int = 30, parallel: bool = False, **_
+    ):
+        assert minmax_ratio > 0, "minmax_ratio must be greater than 0"
+        return super().downsample(
+            *args, n_out=n_out, parallel=parallel, ratio=minmax_ratio
+        )
+
+
+# ------------------ EveryNth Downsampler ------------------
+import math
+
+
+class EveryNthDownsampler(AbstractDownsampler):
     def __init__(self) -> None:
         super().__init__("EveryNth")
 
-    def downsample(
-        self, s: pd.Series, n_out: int, *args, parallel: bool = False
-    ) -> pd.Series:
-        return s[:: max(1, math.ceil(len(s) / n_out))]
+    def _downsample(
+        self, x: Union[np.ndarray, None], y: np.ndarray, n_out: int, **_
+    ) -> np.ndarray:
+        if x is not None:
+            warnings.warn(
+                f"x is passed to downsample method of {self.name}, but is not taken "
+                f"into account by the current implementation of  {self.name} algorithm."
+            )
+        step = max(1, math.ceil(len(y) / n_out))
+        return np.arange(0, len(y), step)
+
+
+# ------------------ Function Downsampler ------------------
+
 
+class FuncDownsampler(AbstractDownsampler):
```

Comment:
Maybe add a docstring + concrete implementation

Reply:
@jvdd what is your opinion?

Reply:
Agree! I think docstrings are missing for most classes tbh